nemo/src/model/types/complex_types.rs

```diff
@@ -133,14 +133,6 @@ impl Deref for TupleConstraint {
     }
 }
 
-impl FromIterator<PrimitiveType> for TupleConstraint {
```

Comment:
I removed this since I found it confusing. I was actually expecting that this would always be turned into `AtLeast` constraints instead of `Exact` constraints, e.g. when using this on Data Sources.
I think one should rather make this explicit and use the `FromIterator` implementation below.